src/test/java/com/faforever/client/patch/DownloadServiceTest.java

```diff
@@ -0,0 +1,78 @@
+package com.faforever.client.patch;
+
+import com.faforever.client.io.DownloadService;
+import com.faforever.client.preferences.PreferencesService;
+import com.faforever.client.test.ServiceTest;
+import javafx.collections.FXCollections;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+public class DownloadServiceTest extends ServiceTest {
```

Comment:
Need a test that the downloadFileWithMirrors rotates through the mirrors properly as well